app/models/nic/interface.rb

```diff
@@ -20,10 +20,28 @@ class Interface < Base
 
     validate :alias_subnet
 
-    delegate :network, :to => :subnet
+    delegate :network, :to => :subnet, :prefix => true
+    delegate :network, :to => :subnet6, :prefix => true
+
+    alias_method :network, :subnet_network
```

Comment:
This will alias `subnet_network` to `network`, so that you can call `self.network` and get `self.subnet_network`, which points to `self.subnet.network`. I know, a lot of indirection :-) The next line does the same for `network6` and `subnet6`.